pyccel/codegen/printing/pycode.py

```diff
@@ -884,10 +884,12 @@ def _print_Slice(self, expr):
                 stop  = stop,
                 step  = step)
 
-    def _print_SetAdd(self, expr):
-        name = self._print(expr.set_variable)
-        args = self._print(expr.add_argument)
-        return f"{name}.add({args})\n"
+    def _print_SetMethod(self, expr):
+        set_var = self._print(expr.set_variable)
+        name = self._print(expr.name)
```

Comment:
Why do you need to call `self._print` to get the Python code representing `expr.name`? Isn't this already a string which does not need to be modified?

Reply:
following the development convention in codegen:

> Each of these _print_X functions should internally call the _print function on each of the elements of the object to obtain strings which can be combined to create a string describing the current object in the target language. It can be tempting to skip some of these _print calls, especially for basic types such as literals

Reply:
Ah ok. Maybe I should modify this documentation. It should read:
> Each of these _print_X functions should internally call the _print function on each of the `PyccelAstNode` elements of the object to obtain strings which can be combined to create a string describing the current object in the target language. It can be tempting to skip some of these _print calls, especially for basic types such as Literals

You are calling this function:
https://github.com/pyccel/pyccel/blob/9c5d09b6c90086f15c5853cd0df1c3a9701a29df/pyccel/codegen/printing/codeprinter.py#L118-L120

which I should probably remove at some point

Reply:
Should I keep it as is or modify it to : 
```suggestion
        name = expr.name
```

Reply:
Okay, I'll make the changes in my next commit.

Reply:
fixed : a44d2a7